packages/nimble/inst/tests/test_utils.R

```diff
@@ -788,49 +788,45 @@ weightedMetricFunc <- function(index, samples, weights, metric, samplesToWeights
 }
 
 test_size <- function(input, verbose = TRUE) {
-    errorMsg <- paste0(ifelse(input$knownProblem, "KNOWN ISSUE: ", ""), "Result does not match ", input$expectPass)
+    if(is.null(input$expectPassWithConst)) input$expectPassWithConst <- input$expectPass
+    if(is.null(input$knownProblem)) input$knownProblem <- FALSE
+    if(is.null(input$knownProblemWithConst)) input$knownProblemWithConst <- input$knownProblem
+
     if(verbose) cat("### Testing", input$name, " with RHS variable ###\n")
-    result <- try(
+    code <- quote({
         m <- nimbleModel(code = input$expr, data = input$data, inits = input$inits)
-    )
-    try(test_that(paste0("Test 1 of size/dimension check: ", input$name),
-                  expect_equal(!is(result, "try-error"), input$expectPass,
-                              info = errorMsg)))
-    if(!is(result, "try-error")) {
-        result <- try(
-            { calculate(m); out <- calculate(m)} )
-        try(test_that(paste0("Test 2 of size/dimension check: ", input$name),
-                      expect_equal(!is(result, "try-error"), input$expectPass,
-                                  info = errorMsg)))
+        calculate(m)  ## Calculates from scratch.
+        calculate(m)  ## Uses cached value.
```

Comment:
I think we can address this behavior later; this PR does preserve existing behavior.